modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/TypeAwareConfigurationInstantiator.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.introspection;
+
+import static org.mule.config.i18n.MessageFactory.createStaticMessage;
+import static org.mule.module.extension.internal.util.IntrospectionUtils.checkInstantiable;
+import static org.mule.util.Preconditions.checkArgument;
+import org.mule.api.MuleRuntimeException;
+import org.mule.extension.introspection.ConfigurationInstantiator;
+
+/**
+ * Implementation of {@link ConfigurationInstantiator} which creates instances
+ * based on a given {@link Class} which is assumed to have a default and public
+ * constructor.
+ *
+ * @since 3.7.0
+ */
+final class TypeAwareConfigurationInstantiator implements ConfigurationInstantiator
+{
+
+    private final Class<?> configurationType;
+
+    /**
+     * Constructor which receives the type to be instantiated on each invocation to
+     * {@link #newInstance()}. This type must be not {@code null}, and have a public default
+     * constructor. Otherwise an {@link IllegalArgumentException} is thrown
```

Comment:
Redundant. "Creates an instance of a given configuration class".
Move the param restriction to param's javadoc